restcomm/restcomm.dao/src/test/java/org/restcomm/connect/dao/mybatis/ApplicationRetrievalTest.java

```diff
@@ -0,0 +1,108 @@
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>
+ *
+ */
+
+package org.restcomm.connect.dao.mybatis;
+
+import junit.framework.Assert;
+import org.apache.ibatis.session.SqlSessionFactory;
+import org.apache.ibatis.session.SqlSessionFactoryBuilder;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.restcomm.connect.commons.dao.Sid;
+import org.restcomm.connect.dao.ApplicationsDao;
+import org.restcomm.connect.dao.entities.Application;
+
+import java.io.FileInputStream;
+import java.io.InputStream;
+import java.util.List;
+
+/**
+ * @author otsakir@gmail.com - Orestis Tsakiridis
+ */
+public class ApplicationRetrievalTest extends DaoTest {
+    private static MybatisDaoManager manager;
+
+    public ApplicationRetrievalTest() {
+        super();
+    }
+
+    @Before
+    public void before() throws Exception {
+        sandboxRoot = createTempDir("applicationRetrievalTest");
+        String mybatisFilesPath = getClass().getResource("/applicationsDao").getFile();
+        setupSandbox(mybatisFilesPath, sandboxRoot);
+
+        String mybatisXmlPath = sandboxRoot.getPath() + "/mybatis_updated.xml";
+        final InputStream data = new FileInputStream(mybatisXmlPath);
+        final SqlSessionFactoryBuilder builder = new SqlSessionFactoryBuilder();
+        final SqlSessionFactory factory = builder.build(data);
+        manager = new MybatisDaoManager();
+        manager.start(factory);
+    }
+
+    @After
+    public void after() throws Exception {
+        manager.shutdown();
+        removeTempDir(sandboxRoot.getAbsolutePath());
+    }
+
+    @Test
+    public void retrieveApplications() {
+        ApplicationsDao dao = manager.getApplicationsDao();
+        List<Application> apps = dao.getApplications(new Sid("ACae6e420f425248d6a26948c17a9e2acf"));
+        Assert.assertEquals(5, apps.size());
+    }
+
+    /**
+     * PN00000000000000000000000000000001 is bound to AP73926e7113fa4d95981aa96b76eca854 (sms)
+     * PN00000000000000000000000000000002 is bound to AP73926e7113fa4d95981aa96b76eca854 (both sms+ussd)
+     * PN00000000000000000000000000000003 is bound to AP00000000000000000000000000000005. The number belong sto other account
+     * AP00000000000000000000000000000006 belongs to other account
+     */
+    @Test
+    public void retrieveApplicationsAndTheirNumbers() {
+        ApplicationsDao dao = manager.getApplicationsDao();
+        List<Application> apps = dao.getApplications(new Sid("ACae6e420f425248d6a26948c17a9e2acf"), true);
+
+        Assert.assertEquals(5, apps.size());
+        Assert.assertNotNull(searchApplicationBySid(new Sid("AP73926e7113fa4d95981aa96b76eca854"),apps).getNumbers());
+        // applications bound with many numbers are property returned
+        Assert.assertEquals("Three (3) numbers should be bound to this application", searchApplicationBySid(new Sid("AP73926e7113fa4d95981aa96b76eca854"),apps).getNumbers().size(), 3);
```

Comment:
i think expectedValue is 2 argument. If 3 is placed in proper place, there is not much requirement on putting custom messag for this assertion, since the default one will be self explanatory

Reply:
correct. fixed it.